src/manifolds/ProductManifold.jl

```diff
@@ -20,6 +20,14 @@ function ProductManifold(manifolds::Manifold...)
     return ProductManifold{𝔽,typeof(manifolds)}(manifolds)
 end
 
+"""
+    getindex(M::ProductManifold, i)
+    M[i]
+
+access the `i`th manifold component from the [`ProductManifold`](@ref) `M`.
+"""
+@inline Base.getindex(M::ProductManifold, i) = M.manifolds[i]
```

Comment:
```suggestion
@inline Base.getindex(M::ProductManifold, i::Integer) = M.manifolds[i]
```
I'd restrict it to integers for now, unless you need indexing with something else? (ranges, colon, etc.)

Reply:
Sure, sounds reasonable, since product manifolds always have a linear, integer indexed order of their manifolds.